src/core/resource-container-helper.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {rethrowAsync} from './rethrow-async';
+
+const AMP_CLASS = 'i-amphtml-element';
```

Comment:
This is all AMP mode specific, so I'm not sure it belongs in `src/core`. For instance, I think it'd be a bug if a Preact component (other than `Slot`, which is an AMP mode component) called any of the functions here.

Reply:
Yeah, I see your point. I actually started with it in the `src/utils/`, but `Slot` is covered by `src/core` restrictions. Is there a way I can modify these restrictions?

Reply:
Moved the `resource-container-helper.js` to `src/utils/`.